presto-spi/src/main/java/io/prestosql/spi/connector/SchemaTablePrefix.java

```diff
@@ -14,64 +14,80 @@
 package io.prestosql.spi.connector;
 
 import java.util.Objects;
+import java.util.Optional;
 
 import static io.prestosql.spi.connector.SchemaUtil.checkNotEmpty;
 
 public class SchemaTablePrefix
 {
-    /* nullable */
-    private final String schemaName;
-    /* nullable */
-    private final String tableName;
+    private final Optional<String> schemaName;
+    private final Optional<String> tableName;
 
     public SchemaTablePrefix()
     {
-        this.schemaName = null;
-        this.tableName = null;
+        this.schemaName = Optional.empty();
+        this.tableName = Optional.empty();
     }
 
     public SchemaTablePrefix(String schemaName)
     {
-        this.schemaName = checkNotEmpty(schemaName, "schemaName");
-        this.tableName = null;
+        this.schemaName = Optional.of(checkNotEmpty(schemaName, "schemaName"));
+        this.tableName = Optional.empty();
     }
 
     public SchemaTablePrefix(String schemaName, String tableName)
     {
-        this.schemaName = checkNotEmpty(schemaName, "schemaName");
-        this.tableName = checkNotEmpty(tableName, "tableName");
+        this.schemaName = Optional.of(checkNotEmpty(schemaName, "schemaName"));
+        this.tableName = Optional.of(checkNotEmpty(tableName, "tableName"));
     }
 
+    /**
+     * @deprecated replaced by {@link SchemaTablePrefix#getSchema()}
+     */
+    @Deprecated
     public String getSchemaName()
+    {
+        return schemaName.orElse(null);
+    }
+
+    public Optional<String> getSchema()
     {
         return schemaName;
     }
 
+    /**
+     * @deprecated replaced by {@link SchemaTablePrefix#getTable()}
+     */
+    @Deprecated
     public String getTableName()
+    {
+        return tableName.orElse(null);
+    }
+
+    public Optional<String> getTable()
     {
         return tableName;
     }
 
     public boolean matches(SchemaTableName schemaTableName)
     {
-        // null schema name matches everything
```

Comment:
Let's leave the comment but change to
```java
// empty prefix matches everything

Reply:
Isn't this already written one line below, but in java?

Anyway, added a comment.